src/LevelZeroShim.hpp

```diff
@@ -0,0 +1,154 @@
+
+            /// @brief Get the number of LevelZero frequency domains of a certain type
+            /// @param [in] domain The domain type being targeted
+            /// @return Accelerator frequency domain count.
+            virtual int frequency_domain_count(unsigned int device_idx, int domain) const = 0;
+            /// @brief Get the LevelZero device actual frequency in MHz
+            /// @param [in] device_idx The index indicating a particular
+            ///        accelerator.
+            /// @param [in] domain The domain type being targeted
+            /// @param [in] domain_idx The index indicating a particular
+            ///        domain of the accelerator.
+            /// @return Accelerator device core clock rate in MHz.
+            virtual double frequency_status(unsigned int device_idx, int domain, int domain_idx) const = 0;
+            /// @brief Get the LevelZero device mininmum frequency in MHz
+            /// @param [in] device_idx The index indicating a particular
+            ///        accelerator.
+            /// @param [in] domain The domain type being targeted
+            /// @param [in] domain_idx The index indicating a particular
+            ///        domain of the accelerator.
+            /// @return Accelerator minimum frequency in MHz.
+            virtual double frequency_min(unsigned int device_idx, int domain, int domain_idx) const = 0;
+            /// @brief Get the LevelZero device maximum frequency in MHz
+            /// @param [in] device_idx The index indicating a particular
+            ///        accelerator.
+            /// @param [in] domain The domain type being targeted
+            /// @param [in] domain_idx The index indicating a particular
+            ///        domain of the accelerator.
+            /// @return Accelerator maximum frequency in MHz.
+            virtual double frequency_max(unsigned int device_idx, int domain, int domain_idx) const = 0;
+
+            /// @brief Get the number of LevelZero engine domains
+            /// @param [in] domain The domain type being targeted
+            /// @return Accelerator engine domain count.
+            virtual int engine_domain_count(unsigned int device_idx, int domain) const = 0;
+            /// @brief Get the LevelZero device active time in microseconds
+            /// @param [in] device_idx The index indicating a particular
+            ///        accelerator.
+            /// @param [in] domain The domain type being targeted
+            /// @param [in] domain_idx The index indicating a particular
+            ///        domain of the accelerator.
+            /// @return Accelerator active time in microseconds.
+            virtual uint64_t active_time(unsigned int device_idx, int domain, int domain_idx) const = 0;
+            /// @brief Get the LevelZero device timestamp for the active time value in microseconds
+            /// @param [in] device_idx The index indicating a particular
```

Comment:
```suggestion
            /// @param [in] device_idx The L0 index indicating a particular
```

In general, spelling out what's in the context for L0 vs. GEOPM would help.

Reply:
Updated to be l0_device_idx and to make it clear this is a l0 device, not a GEOPM Device